### DISCUSSION TITLE
Extremely high memory usage when opening a large project

### DISCUSSION THREAD

**Discussion by koby08** [REditorSupport/languageserver#599] [General]
Hi,

I've been trying to use the language server in a quite big private library (~10MB of code), and the language server indeed indexed all the R files. However, its memory consumption reached ~12GB.

Is there a way I could minimize the footprint?

Thanks

**Comment by renkun-ken** [comment]
languageserver currently will parse all R files in `R` folder in the workspace if it contains `NAMESPACE` file, indicating that it is an R package. Is it your case?

**Reply by Ohmegas** [reply]
Yes, it's a quite big library. In RStudio I didn't see any issues or high memory usage, but I really want to give vscode try

**Reply by renkun-ken** [reply]
Would you like to give us some simple numbers to describe how large is the package? Number of files in `R`, average file size, etc? Looks like we should give up parsing all files in `R` if we find the folder contains too many files, or there might be smarter behavior so that we only parse a subset of files.

**Reply by Ohmegas** [reply]
Sure - few hundreds of files, ~10mb of code, pretty much evenly distributed between them..

Hmm, it is a relatively big project I assume, but I dont really understand how come loading 10mb of code becomes 12gb of RAM? Any idea? RStudio for comparison seems to work well in this project, and takes ~100mb of RAM

**Comment by Ohmegas** [comment]
I wrote a small patch that instead of relying on the parsed files that are loaded in `workspace$load_all` - it extracts the srcref of the function that was received in a definition request, and then parses the relevant file adhoc, and then tries to resolve the definition.

It works pretty well, but still using a hacky implementation.

What do you think about such a feature for the server? (Of course with a proper implementation)

**Reply by koby08** [reply]
@renkun-ken Any feedback on this?

**Comment by Tal500** [comment]
I'm working at the same company this issue happens. I've opened a detailed issue and suggestion in #618
